SNT/Utils.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace Torch.SNT
+{
+    internal class Utils
+    {
+        public static int GetTotalLength(ReadOnlySpan<int> dimensions, int startIndex = 0)
+        {
+            if (dimensions.Length == 0)
+            {
+                return 0;
+            }
+
+            int product = 1;
+            for (int i = startIndex; i < dimensions.Length; i++)
+            {
+                if (dimensions[i] < 0)
```

Comment:
Can we avoid this check by using unsinged integers for the dimensions?

Reply:
I think this is difficult to do, unless we can convince the SNT folks to use unsigned ints in their API. (Torch is actually using long BTW)